### PR TITLE
Feature/brooming

### LINKED ISSUE #346: Built year is null error
Stacktrace:

```
System.InvalidCastException: Column 'built_year' is null.
   at Npgsql.ThrowHelper.ThrowInvalidCastException_NoValue(FieldDescription field)
   at Npgsql.NpgsqlDataReader.GetFieldValue[T](Int32 ordinal)
   at Npgsql.NpgsqlDataReader.GetDateTime(Int32 ordinal)
   at FunderMaps.Data.Repositories.BuildingRepository.MapFromReader(DbDataReader reader, Int32 offset) in C:\Users\thoma\Programming\Laixer\FunderMaps\src\FunderMaps.Data\Repositories\BuildingRepository.cs:line 46
   at FunderMaps.Data.Repositories.AddressRepository.MapFromReader(DbDataReader reader, Boolean fullMap, Int32 offset) in C:\Users\thoma\Programming\Laixer\FunderMaps\src\FunderMaps.Data\Repositories\AddressRepository.cs:line 30
   at FunderMaps.Data.Repositories.AddressRepository.GetBySearchQueryAsync(String query, INavigation navigation)+MoveNext() in C:\Users\thoma\Programming\Laixer\FunderMaps\src\FunderMaps.Data\Repositories\AddressRepository.cs:line 230
   at FunderMaps.Data.Repositories.AddressRepository.GetBySearchQueryAsync(String query, INavigation navigation)+MoveNext() in C:\Users\thoma\Programming\Laixer\FunderMaps\src\FunderMaps.Data\Repositories\AddressRepository.cs:line 228
   at FunderMaps.Data.Repositories.AddressRepository.GetBySearchQueryAsync(String query, INavigation navigation)+MoveNext() in C:\Users\thoma\Programming\Laixer\FunderMaps\src\FunderMaps.Data\Repositories\AddressRepository.cs:line 228
   at FunderMaps.Data.Repositories.AddressRepository.GetBySearchQueryAsync(String query, INavigation navigation)+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()
   at AutoMapper.IMapperExtensions.MapAsync[TDestination,TEntity](IMapper mapper, IAsyncEnumerable`1 enumerable, CancellationToken token) in C:\Users\thoma\Programming\Laixer\FunderMaps\src\FunderMaps.Core\Extensions\IMapperExtensions.cs:line 41
   at AutoMapper.IMapperExtensions.MapAsync[TDestination,TEntity](IMapper mapper, IAsyncEnumerable`1 enumerable, CancellationToken token) in C:\Users\thoma\Programming\Laixer\FunderMaps\src\FunderMaps.Core\Extensions\IMapperExtensions.cs:line 41
   at FunderMaps.WebApi.Controllers.Geocoder.AddressController.GetAllSuggestionAsync(AddressSearchDto input) in C:\Users\thoma\Programming\Laixer\FunderMaps\src\FunderMaps.WebApi\Controllers\Geocoder\AddressController.cs:line 62
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Logged|12_1(ControllerActionInvoker invoker)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Builder.Extensions.UsePathBaseMiddleware.Invoke(HttpContext context)
   at FunderMaps.AspNetCore.ErrorMessaging.CustomExceptionHandlerMiddleware`1.ProcessAsync(HttpContext context, Task task) in C:\Users\thoma\Programming\Laixer\FunderMaps\src\FunderMaps.AspNetCore\ErrorMessaging\CustomExceptionHandlerMiddleware.cs:line 81
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

### PR DESCRIPTION
**Pull request by yorickdewid**
**What issue does this PR address?**
Various small bugs, fixes, improvements. See full change log
Fixes #346 

**Does this PR introduce a breaking change?**
Nope

### CODE REVIEW

File: src/FunderMaps.Core/Entities/Building.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public Building()
         /// <summary>
         ///     Built year.
         /// </summary>
-        public DateTime BuiltYear { get; set; }
+        public DateTime? BuiltYear { get; set; }
```
Comment (tabeckers) [inline src/FunderMaps.Core/Entities/Building.cs:32]: I noticed a recurring bug where built_year was null in our backend, see #346. Is this the fix for that issue?

Comment (yorickdewid) [reply src/FunderMaps.Core/Entities/Building.cs:32]: Jup, shoud have linked

File: src/FunderMaps.Core/Types/Products/AnalysisProductType.cs [hunk 2]
```diff
@@ -8,37 +8,39 @@ public enum AnalysisProductType
...
+        Complete = 4,
 
         /// <summary>
-        ///     Represents descriptive texts for a given building.
+        ///     Represents all data about the risks for a given building.
         /// </summary>
-        BuildingDescription,
+        Risk = 5,
 
+#if DEBUGY || USE_BUILDINGDESCRIPTION // NOTE: This is a future feature.
```
Comment (tabeckers) [inline src/FunderMaps.Core/Types/Products/AnalysisProductType.cs:39]:
```suggestion
#if DEBUG || USE_BUILDINGDESCRIPTION // NOTE: This is a future feature.
```

File: src/FunderMaps.WebApi/Controllers/Report/InquiryController.cs [hunk 3]
```diff
@@ -216,7 +216,10 @@ public async Task<IActionResult> SetStatusReviewAsync(int id, StatusChangeDto in
 
             // Act.
             await _inquiryRepository.SetAuditStatusAsync(inquiry.Id, inquiry);
-            await _notificationService.NotifyByEmailAsync(new string[] { "info@example.com" }, input.Message);
+            await _notificationService.NotifyByEmailAsync(
+                address: new string[] { "info@example.com" }, // TODO:
+                content: input.Message,
+                subject: "FunderMaps - Rapportage ter review");
```
Comment (tabeckers) [inline src/FunderMaps.WebApi/Controllers/Report/InquiryController.cs:222]: Place in constants, don't hardcode

File: src/FunderMaps.WebApi/Controllers/Report/InquiryController.cs [hunk 4]
```diff
@@ -237,7 +240,10 @@ public async Task<IActionResult> SetStatusRejectedAsync(int id, StatusChangeDto
 
             // Act.
             await _inquiryRepository.SetAuditStatusAsync(inquiry.Id, inquiry);
-            await _notificationService.NotifyByEmailAsync(new string[] { "info@example.com" }, input.Message);
+            await _notificationService.NotifyByEmailAsync(
+                address: new string[] { "info@example.com" }, // TODO:
+                content: input.Message,
+                subject: "FunderMaps - Rapportage afgekeurd");
```
Comment (tabeckers) [inline src/FunderMaps.WebApi/Controllers/Report/InquiryController.cs:246]: Same

File: src/FunderMaps.WebApi/Controllers/Report/InquiryController.cs (hunk 3 above)
Comment (yorickdewid) [reply src/FunderMaps.WebApi/Controllers/Report/InquiryController.cs:222]: Not in a constant, but in the resource table.

File: src/FunderMaps.Data/Repositories/IncidentRepository.cs [hunk 5]
```diff
@@ -266,7 +266,6 @@ UPDATE  report.incident
                             document_file = @document_file,
                             note = @note,
                             internal_note = @internal_note,
-                            contact = @email,
```
Comment (tabeckers) [inline src/FunderMaps.Data/Repositories/IncidentRepository.cs:269]: Why was this removed?

Comment (yorickdewid) [reply src/FunderMaps.Data/Repositories/IncidentRepository.cs:269]: Contact is a foreign key, which can only be updated when the new entry exists. There should be no necessarily to update the contact link, but only ever the contact.

**Review by tabeckers — CHANGES_REQUESTED** [review]
See comments

File: src/FunderMaps.WebApi/Controllers/Report/InquiryController.cs (hunk 3 above)
Comment (yorickdewid) [reply src/FunderMaps.WebApi/Controllers/Report/InquiryController.cs:222]: See #353

File: src/FunderMaps.WebApi/Controllers/Report/InquiryController.cs (hunk 4 above)
Comment (yorickdewid) [reply src/FunderMaps.WebApi/Controllers/Report/InquiryController.cs:246]: See #353

**Review by tabeckers — APPROVED** [review]
